nipoppy_cli/nipoppy/config/pipeline.py

```diff
@@ -4,36 +4,75 @@
 from pathlib import Path
 from typing import Optional, Sequence
 
-from pydantic import ConfigDict, model_validator
+from pydantic import ConfigDict, Field, model_validator
 
 from nipoppy.config.container import ModelWithContainerConfig
 
 
 class PipelineConfig(ModelWithContainerConfig):
-    """Model for workflow configuration."""
-
-    DESCRIPTION: Optional[str] = None
-    CONTAINER: Optional[Path] = None
-    URI: Optional[str] = None
-    DESCRIPTOR: Optional[dict] = None
-    DESCRIPTOR_FILE: Optional[Path] = None
-    INVOCATION: Optional[dict] = None
-    INVOCATION_FILE: Optional[Path] = None
-    PYBIDS_IGNORE: list[re.Pattern] = []
-    TRACKER_CONFIG: dict[str, list[str]] = {}
+    """Model for processing pipeline configuration."""
```

Comment:
I am hesitant using the term `model` here. It's very overloaded and creates confusion. I presume this could be called `template` or `recipe` ?

Reply:
Schema maybe?